store_test.go

```diff
@@ -113,10 +116,10 @@ func TestStoreSync(t *testing.T) {
 	n := newTransport("secret-123", opt)
 	s := newStoreInternal(n, time.Second, time.Second)
 
-	if len(s.featureGates) != 1 {
+	if len(s.getAllGates()) != 1 {
```

Comment:
Is this safe (I'm not sure)? The getAllGates() method is returning a reference to a map that's concurrently accessed.

Reply:
not sure either but I think race detector is passing because this is only checked here and not used anywhere else. I just changed the `getAllGates()` method to return the count now so that should be safe

Reply:
I bet you're right. A race condition has to happen at runtime for the race detector to go off.